src/pages/home/report/ReportActionItemParentAction.js

```diff
@@ -27,77 +30,127 @@ const propTypes = {
+                                displayAsGroup={false}
+                                isMostRecentIOUReportAction={false}
+                                shouldDisplayNewMarker={ancestor.shouldDisplayNewMarker}
+                                index={props.index}
+                            />
+                            {!shouldHideThreadDividerLine && <View style={[styles.threadDividerLine]} />}
+                        </OfflineWithFeedback>
+                    );
+                })}
             </View>
-            {!props.shouldHideThreadDividerLine && <View style={[styles.threadDividerLine]} />}
-        </OfflineWithFeedback>
+        </>
     );
 }
 
 ReportActionItemParentAction.defaultProps = defaultProps;
 ReportActionItemParentAction.propTypes = propTypes;
 ReportActionItemParentAction.displayName = 'ReportActionItemParentAction';
 
+/**
+ * @param {Object} [reportActions]
+ * @returns {Object|undefined}
+ */
+const reportActionsSelector = (reportActions) =>
+    reportActions &&
+    _.map(reportActions, (reportAction) => ({
+        errors: lodashGet(reportAction, 'errors', []),
+        pendingAction: lodashGet(reportAction, 'pendingAction'),
+        message: reportAction.message,
+        created: reportAction.created,
+    }));
+
+/**
+ * @param {Object} [report]
+ * @returns {Object|undefined}
+ */
+const reportSelector = (report) =>
+    report && {
+        reportID: report.reportID,
+        isHidden: report.isHidden,
+        errorFields: {
+            createChat: report.errorFields && report.errorFields.createChat,
+            addWorkspaceRoom: report.errorFields && report.errorFields.addWorkspaceRoom,
+        },
+        pendingFields: {
+            createChat: report.pendingFields && report.pendingFields.createChat,
+            addWorkspaceRoom: report.pendingFields && report.pendingFields.addWorkspaceRoom,
+        },
+        statusNum: report.statusNum,
+        stateNum: report.stateNum,
+        lastReadTime: report.lastReadTime,
+        // Other important less obivous properties for filtering:
+        parentReportActionID: report.parentReportActionID,
+        parentReportID: report.parentReportID,
+        isDeletedParentAction: report.isDeletedParentAction,
+    };
+
 export default compose(
     withWindowDimensions,
     withLocalize,
     withOnyx({
-        report: {
-            key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+        // We should subscribe all reports and report actions here to dynamic update when any parent report action is changed
+        allReportActions: {
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            selector: reportActionsSelector,
         },
-        parentReportActions: {
-            key: ({parentReportID}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
-            canEvict: false,
+        allReports: {
+            key: ONYXKEYS.COLLECTION.REPORT,
+            selector: reportSelector,
```

Comment:
Why do we need to make a selection here even though the props are not used?

Reply:
I think using `memo` will be ideal. We already use it for several components, which fits what we are trying to do here.